indexd/utils.py

```diff
@@ -9,7 +9,17 @@
 
 logger = logging.getLogger(__name__)
 
-
+def handle_error(resp):
+    if 400 <= resp.status_code < 600:
+        try:
+            json = resp.json()
+            resp.reason = json["error"]
```

Comment:
This will be safer and prevent our error handling code from need more error handling. 
```suggestion
            resp.reason = json.get("error")
```